contracts/staking/TribalChiefSyncExtension.sol

```diff
@@ -0,0 +1,70 @@
+pragma solidity ^0.8.0;
+
+import "./TribalChiefSyncV2.sol";
+
+/**
+    @title TribalChiefSyncExtension supports multiple auto-reward-distributors
+    @author joeysantoro
+
+    @notice Include the autoRewardsDistributors with the TribalChiefSyncV2 permissionless functions. 
+    The TribalChiefSyncExtension will sync the autoRewardsDistributors after completing the main sync
+ */
+contract TribalChiefSyncExtension {
+    TribalChiefSyncV2 public immutable tribalChiefSync;
+
+    constructor(TribalChiefSyncV2 _tribalChiefSync) {
+        tribalChiefSync = _tribalChiefSync;
+    }
+
+    modifier update(IAutoRewardsDistributor[] calldata distributors) {
+        _;
+        unchecked {
+            for (uint256 i = 0; i < distributors.length; i++) {
+                distributors[i].setAutoRewardsDistribution();
+            }
+        }
+    }
+
+    /// @notice Sync a rewards rate change automatically using pre-approved map
+    function autoDecreaseRewards(IAutoRewardsDistributor[] calldata distributors) external update(distributors) {
+        tribalChiefSync.autoDecreaseRewards();
+    }
+
+    /// @notice Sync a rewards rate change
+    function decreaseRewards(uint256 tribePerBlock, bytes32 salt, IAutoRewardsDistributor[] calldata distributors) external update(distributors) {
+        tribalChiefSync.decreaseRewards(tribePerBlock, salt);
+    }
+
+    /// @notice Sync a pool addition
+    function addPool(
+        uint120 allocPoint, 
+        address stakedToken, 
+        address rewarder, 
+        TribalChiefSyncV2.RewardData[] calldata rewardData, 
+        bytes32 salt,
+        IAutoRewardsDistributor[] calldata distributors
+    ) external update(distributors) {
+        tribalChiefSync.addPool(allocPoint, stakedToken, rewarder, rewardData, salt);
```

Comment:
Do we need to update the distributors after adding the pool?

Reply:
thats what update modifier does and yes we need to